server/src/main/java/org/opensearch/index/engine/NRTReplicationEngine.java

```diff
@@ -337,6 +337,7 @@ public void refresh(String source) throws EngineException {
 
     @Override
     public boolean maybeRefresh(String source) throws EngineException {
+        ensureOpen();
```

Comment:
This check can be omitted as the call to `maybeRefresh()` performs this check.

Reply:
These are two separate methods both named `ensureOpen`, the one added here checks to ensure the engine is open, defined in Engine.java.  The one from readerManager.maybeRefresh() is on the ReferenceManager which ensures the class it is refcounting, the reader, is not closed.

Reply:
My bad, you are right.